### PR TITLE
PandasTransform - Ready for review

### PR DESCRIPTION
**Pull request by rcrowe-google**
[Project Proposal](https://github.com/tensorflow/tfx-addons/blob/main/proposals/20220513-pandas_transform.md)

This is an initial working version of the PandasTransform component.  One feature not in the proposal is that it checks to see if a [Beam custom component](https://www.tensorflow.org/tfx/guide/custom_function_component) (BaseBeamComponent) is supported in the loaded version of TFX, and uses it if it is.

### CODE REVIEW

**Comment by rcrowe-google** [comment]
There's something failing in yapf, but other than that I think this is ready for review.

File: tfx_addons/pandas_transform/component.py
Comment (casassg) [inline tfx_addons/pandas_transform/component.py:144]: nit: should we add beam_pipeline_args somehow?

**Review by casassg — COMMENTED** [review]
You can run YAPF locally by installing `pre-commit install` -- https://github.com/tensorflow/tfx-addons/blob/main/CONTRIBUTING.md#development-tips

Also seems a CI test is failing, not sure why, can take a look

File: tfx_addons/pandas_transform/component.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py:144]: They'll be added for TFX >= 1.8.0.  I haven't really thought about how to add them for earlier versions.

Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py:144]: Also I really need to add more tests and at least one example.

**Comment by rcrowe-google** [comment]
Yapf is still complaining about something, but I can't quite see what it is.

**Comment by casassg** [comment]
Also, @rcrowe-google would be curious if you want to add more context to Contributing.md file. Would be nice to double validate that pre-commit stuff works fine as we get more contributors.

File: tfx_addons/pandas_transform/component.py
Comment (casassg) [inline tfx_addons/pandas_transform/component.py:244]: nit: Should we try to check if its a GCS file and download it if so?

File: tfx_addons/version.py
Comment (casassg) [inline tfx_addons/version.py:71]: nit: Should we add a cap to pandas? (aka <2.0) to avoid breaking if there's a new major version of pandas?

File: tfx_addons/pandas_transform/README.md
Comment (casassg) [inline tfx_addons/pandas_transform/README.md:64]: nit: Let's update also CODEOWNERS btw

**Review by casassg — APPROVED** [review]

File: examples/pandas_transform/pandas_transform_example.ipynb
Comment (rcrowe-google) [reply examples/pandas_transform/pandas_transform_example.ipynb:216]: I went back and forth on that.  It does add bloat, but it's nice to see some output without having to run it.  I could go either way.

File: tfx_addons/pandas_transform/component.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py:244]: It looks like that requires some parsing of the URI to extract the bucket name.  I'm not finding any nice clean way to do that.

File: tfx_addons/version.py
Comment (rcrowe-google) [reply tfx_addons/version.py:71]: Good idea, added.

File: examples/pandas_transform/pandas_transform_example.ipynb
Comment (casassg) [reply examples/pandas_transform/pandas_transform_example.ipynb:216]: lets leave as is, we should try to find a better standard for this but not needed for this specific change

File: tfx_addons/pandas_transform/component.py
Comment (casassg) [reply tfx_addons/pandas_transform/component.py]: to be fair, we use https://github.com/tensorflow/tfx/blob/b92695748e4ea35164f7791fc3abed0d1d4679e2/tfx/utils/io_utils.py#L38 for some internal use cases so it may work for this as well

File: CODEOWNERS
Comment (hanneshapke) [inline CODEOWNERS:50]: @casassg Do we need to lint for blank lines are the end of files?

File: tfx_addons/pandas_transform/__init__.py
Comment (hanneshapke) [inline tfx_addons/pandas_transform/__init__.py:19]: @rcrowe-google Doesn't the number should match the version in `Release.md`?

File: tfx_addons/pandas_transform/component.py
Comment (hanneshapke) [inline tfx_addons/pandas_transform/component.py:140]: @rcrowe-google Nice pattern.

File: tfx_addons/pandas_transform/component.py
Comment (hanneshapke) [inline tfx_addons/pandas_transform/component.py:152]: @rcrowe-google Does this need an update before merging?

File: tfx_addons/pandas_transform/component.py
Comment (hanneshapke) [inline tfx_addons/pandas_transform/component.py:127]: @rcrowe-google Does this need an update before merging?

**Review by hanneshapke — APPROVED** [review]

**Comment by hanneshapke** [comment]
@rcrowe-google I left some questions in the PR, but all are non-blocking.

File: CODEOWNERS
Comment (casassg) [reply CODEOWNERS:50]: we do lint for that in python

**Comment by casassg** [comment]
@rcrowe-google this is ready for merging, you may need to rebase to get CI fixes in main (or you can just merge and it should pick it up there)

File: tfx_addons/pandas_transform/component.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py]: I added beam_pipeline_args for both the new and old versions:

>beam_pipeline_args: A string with the argv options for creating a Beam pipeline.
    Note that this is a string, not a list.  It will be split on spaces to create
    a list. If running TFX >= 1.8.0, if beam_pipeline_args are specified they will
    override the pipeline beam args.

File: tfx_addons/pandas_transform/__init__.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/__init__.py:19]: Doh! Yes, it should.  Fixed.  Thanks!

File: tfx_addons/pandas_transform/component.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py]: Good idea, I sorta forgot about that whole thing.

File: tfx_addons/pandas_transform/component.py
Comment (rcrowe-google) [reply tfx_addons/pandas_transform/component.py]: I clarified it here and in the other one.  I thought about turning off the linting for it, but I think explaining it is better.

**Pull request merged**
